src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -210,7 +210,11 @@ impl<'a> JobQueue<'a> {
             // being a compiled package
             Dirty => {
                 self.printed.insert(pkg);
-                try!(config.shell().status("Compiling", pkg));
+                //check if documenting or compiling
+                match doc {
+                    &true => try!(config.shell().status("Documenting", pkg)),
```

Comment:
Wouldn't it be better to just keep `printed` and just do both printings at once?